includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -1048,6 +1056,86 @@ public function getAsJson() {
 		$this->setResponseData( $result, '', self::SIMPLE_JSON_VARNISH_CACHE_EXPIRATION );
 	}
 
+	/**
+	 * Get some or all of the article top level sections as an array
+	 *
+	 * Note: If sections 1, 2 and 3 are requested, but section 3 is a subsection, the returned
+	 * array will only have two items: section 1 & 2 (section 2 containing subsection 3).
+	 * All top level sections will contain all of their subsections, regardless of how many
+	 * were requested. So if sections 1, 2 and 3 are requested, and section 2 contains 3 subsections,
+	 * section two will be returned with all 3 subsections.
+	 *
+	 * @param mixed $sectionsToGet Array of section numbers (could need URL decoding) or the string "all"
+	 * @param ParserOutput $parsedArticle
+	 * @param Article $article
+	 * @param Title $title
+	 * @return array
+	 * @throws BadRequestApiException
+	 */
+	private function getArticleSections( $sectionsToGet, ParserOutput $parsedArticle, Article $article, $title ) {
+		$content = [];
+		$articleAsWikiText = $article->getContent();
+		$parsedSections = $parsedArticle->getSections();
+
+		// decode strings like "%5B1,2,3%5D" or "[1,2,3]"
+		if ( is_string( $sectionsToGet ) ) {
```

Comment:
It at this point `$sectionsToGet` is 'all', it will get `json_decode`-d and will end being `null` after this block;